invokeai/app/extensions/extension_manager.py

```diff
@@ -0,0 +1,39 @@
+import pathlib
+from typing import List
+
+from ...backend.util.logging import getLogger
+from ..services.config import get_invokeai_config
+
+
+class InvokeAIExtensionManager():
+    def __init__(self) -> None:
+        self.logger = getLogger('Extension Manager')
+        self.config = get_invokeai_config()
+        self.community_nodes_dir = self.config.nodes_path
+        assert self.community_nodes_dir.is_dir()
+
+    def load_extensions(self) -> List:
+        loaded_extensions = []
+        for extension in self.community_nodes_dir.iterdir():
+            if extension.is_dir() and not extension.name.startswith('__'):
+                # Search for py files that are not named __init__.py in extensions root directory
+                py_files = list(extension.glob('*.py'))
+                py_files = [
+                    file for file in py_files if not file.name == "__init__.py"]
+
+                # If no py files are found, extension is not loaded.
+                if len(py_files) == 0:
+                    self.logger.warn(
+                        f'Extension: "{extension.name}" failed to load. No nodes found.')
```

Comment:
Are we loading more than just nodes?

Reply:
No. We will only be loading the nodes. If there are supporting files, the nodes will have to draw them directly.

Reply:
So the nodes we load here still have the same, uh, filesystem location? I know the files don't move, but when a node references like `"./some_resource.json"`, does the path voodoo cast by the extension manager interfere?

Reply:
They will need to provide abs paths. We handle everything with pathlib and we'd recommend they do too so there's no clashes.

But I'll try to see if I can implement some localized namespace system for those folders that will allow loading of relative files. Honestly, it might work .. I just didn't try yet. Will give it a shot in just a bit.

I just updated the script to only load node files that actually have Nodes in them.